main/src/ca/uwaterloo/flix/language/errors/RedundancyError.scala

```diff
@@ -540,6 +540,33 @@ object RedundancyError {
 
   }
 
+  /**
+    * An error raised to indicate that a supercast is redundant.
+    *
+    * @param loc the source location of the upcast.
+    */
+  case class RedundantSupercast(loc: SourceLocation) extends RedundancyError {
+
+    def summary: String = "Redundant supercast. The expression already has the expected type."
```

Comment:
Could be nice to see the type we're upcasting to

Reply:
Isn't this explained in the `message` function?